src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -378,9 +403,28 @@ private Task updateAdvancedParameters(ModelSimulation simulation, AdvancedParame
          return _advancedParameterMapper.MapToModel(snapshotAdvancedParameters, simulation as PopulationSimulation);
       }
 
-      private Task updateParameters(ModelSimulation simulation, LocalizedParameter[] snapshotParameters)
+      private async Task updateParameters(ModelSimulation simulation, LocalizedParameter[] snapshotParameters, WithIdRepository withIdRepository)
       {
-         return _parameterMapper.MapLocalizedParameters(snapshotParameters, simulation.Model.Root);
+         await _parameterMapper.MapLocalizedParameters(snapshotParameters, simulation.Model.Root);
+
+         //The parameter mapper does not update the original building block in the simulation. We need to make sure we are keeping these values in sync so 
+         // that update and commit will work as expected (or building block difference )
+         var allParameters = _containerTask.CacheAllChildren<IParameter>(simulation.Model.Root);
+
+         snapshotParameters?.Each(x =>
+         {
+            var parameter = allParameters[x.Path];
+            //Somehow the localized parameter path was edited by hand or does not exist anymore. Warning was already shown during mapping. 
+            if (parameter == null)
+               return;
```

Comment:
what is the use case when something like this can happen?

Reply:
And shouldn't we produce a warning if this happens?

Reply:
See the comment right above the selected line:
1/ happens when snapshot was modified by hand. It maybe the BB creation results in this parameter gone (e.g renamed etc)
2/ warning already shown earlier

For once .all questions answered the comment;-)

Reply:
ok